libs/MeshKernel/src/Mesh2DToCurvilinear.cpp

```diff
@@ -217,80 +226,128 @@ UInt Mesh2DToCurvilinear::ComputeNeighbouringFaceNodes(const UInt face,
         }
     }
     auto nextEdgeIndexInNewFace = edgeIndexInNewFace + 1;
-    nextEdgeIndexInNewFace = nextEdgeIndexInNewFace == 4 ? 0 : nextEdgeIndexInNewFace;
+    nextEdgeIndexInNewFace = nextEdgeIndexInNewFace == geometric::numNodesInQuadrilateral ? 0 : nextEdgeIndexInNewFace;
     const auto nextEdgeInNewFace = m_mesh.m_facesEdges[newFace][nextEdgeIndexInNewFace];
     const auto firstCommonNode = m_mesh.FindCommonNode(edgeIndex, nextEdgeInNewFace);
     const auto firstOtherNode = OtherNodeOfEdge(m_mesh.GetEdge(nextEdgeInNewFace), firstCommonNode);
     const auto iFirstOtherNode = m_i[firstCommonNode] + m_directionsDeltas[d][0];
     const auto jFirstOtherNode = m_j[firstCommonNode] + m_directionsDeltas[d][1];
 
     auto previousEdgeIndexInNewFace = edgeIndexInNewFace - 1;
-    previousEdgeIndexInNewFace = previousEdgeIndexInNewFace == -1 ? 3 : previousEdgeIndexInNewFace;
+    previousEdgeIndexInNewFace = previousEdgeIndexInNewFace == -1 ? geometric::numNodesInQuadrilateral - 1 : previousEdgeIndexInNewFace;
     const auto previousEdgeInNewFace = m_mesh.m_facesEdges[newFace][previousEdgeIndexInNewFace];
     const auto secondCommonNode = m_mesh.FindCommonNode(edgeIndex, previousEdgeInNewFace);
     const auto secondOtherNode = OtherNodeOfEdge(m_mesh.GetEdge(previousEdgeInNewFace), secondCommonNode);
     const auto iSecondCommonNode = m_i[secondCommonNode] + m_directionsDeltas[d][0];
     const auto jSecondCommonNode = m_j[secondCommonNode] + m_directionsDeltas[d][1];
 
-    const auto invalid = (m_i[firstOtherNode] != missing::intValue && m_i[firstOtherNode] != iFirstOtherNode) ||
-                         (m_j[firstOtherNode] != missing::intValue && m_j[firstOtherNode] != jFirstOtherNode) ||
-                         (m_i[secondOtherNode] != missing::intValue && m_i[secondOtherNode] != iSecondCommonNode) ||
-                         (m_j[secondOtherNode] != missing::intValue && m_j[secondOtherNode] != jSecondCommonNode);
+    const auto invalid = m_i[firstOtherNode] != missing::intValue && m_i[firstOtherNode] != iFirstOtherNode ||
+                         m_j[firstOtherNode] != missing::intValue && m_j[firstOtherNode] != jFirstOtherNode ||
+                         m_i[secondOtherNode] != missing::intValue && m_i[secondOtherNode] != iSecondCommonNode ||
+                         m_j[secondOtherNode] != missing::intValue && m_j[secondOtherNode] != jSecondCommonNode;
```

Comment:
I think you may need brackets around the and-ed or or-ed sub-expressions a && b || c && d
If this 
(a && b ) || (c && d ) 
or
a && (b || c) && d

I thikn the original brackets were necessary